stacks-signer/src/runloop.rs

```diff
@@ -138,21 +138,96 @@ pub struct RunLoop<C> {
     pub transactions: Vec<StacksTransaction>,
     /// This signer's ID
     pub signer_id: u32,
+    /// The signer set for this runloop
+    pub signer_set: Option<u32>,
+    /// The index into the signers list of this signer's key (may be different from signer_id)
+    pub signer_slot_id: Option<u32>,
     /// The IDs of all signers partipating in the current reward cycle
     pub signer_ids: Vec<u32>,
     /// The stacks addresses of the signers participating in the current reward cycle
     pub signer_addresses: Vec<StacksAddress>,
 }
 
 impl<C: Coordinator> RunLoop<C> {
+    /// Get and store the signer set assignment for this runloop.
+    /// This assigns the runloop to the _next_ reward cycle, not the current one.
+    /// Returns (signer-set, signer-slot-id)
+    fn get_or_set_signer_info(&mut self) -> Result<(u32, u32), ClientError> {
+        match (self.signer_set.as_ref(), self.signer_slot_id.as_ref()) {
+            (Some(signer_set), Some(signer_slot_id)) => {
+                return Ok((*signer_set, *signer_slot_id));
+            }
+            (_, _) => {}
+        }
+
+        let signer_set = if let Some(signer_set) = self.signer_set.as_ref() {
+            *signer_set
+        } else {
+            let rc = u32::try_from(self.stacks_client.get_current_reward_cycle()?)
+                .expect("FATAL: reward cycle exceeds u32::MAX")
```

Comment:
I thought the reward cycle max was u64? Why down cast it here? I would down cast the modulus component u32::try_from(rc %2) as this will always fit a u32

Reply:
Yeah, I'll do that